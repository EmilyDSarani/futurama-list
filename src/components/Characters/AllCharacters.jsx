//Will figure out the list magic here
//probably will need a link to the character detail page
import { Link } from 'react-router-dom'
import styles from './Character.css'

export default function AllCharacters({ characters }) {
    return (
        <ul aria-label="characters" className={styles.biglist}>
            {characters.map((character, i) =>(
                <li className={styles.list} key={character.Name + i}>
                    <Link to={`/characters/${character.Name}`}>
                            <img src={character.PicUrl}
                                alt={character.Name} />
                            <button><h1>{character.Name}</h1></button>
                    </Link>        
                        
                </li>
            ))}
        <div className={styles.bg}></div>
        <div className={styles.bg2}></div>
        <div className={styles.bg3}></div>
        <div className={styles.content}> </div>
        </ul>
    )
}
