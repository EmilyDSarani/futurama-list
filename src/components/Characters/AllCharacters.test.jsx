import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router';
import AllCharacters from './AllCharacters'

it('should render characters', () => {
    const { container } = render(
    <MemoryRouter> 
    <AllCharacters characters={[    
    {
    name: 'Sarani',
    id: 1,
    image:'https://place-puppy.com/'
}
    ]}/>
    </MemoryRouter> 
    );
    expect(container).toMatchSnapshot();
})