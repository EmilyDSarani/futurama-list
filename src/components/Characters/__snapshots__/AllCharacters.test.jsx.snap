// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render characters 1`] = `
<div>
  <ul
    aria-label="characters"
    class="biglist"
  >
    <li
      class="list"
    >
      <a
        href="/characters/undefined"
      >
        <img
          class="image"
        />
        <button>
          <h1 />
        </button>
      </a>
    </li>
    <div
      class="bg"
    />
    <div
      class="bg2"
    />
    <div
      class="bg3"
    />
    <div
      class="content"
    >
       
    </div>
  </ul>
</div>
`;
